### PR TITLE
Add sparse GQA decode varlen mask operator

### PR DESCRIPTION
**Pull request by 2386225895t**
新增GQA_Decoder_Varlen_Mask算子，测试用例及性能结果如下：
image

### CODE REVIEW

**Review by LLMZhangYC — DISMISSED** [review]
lgtm

File: examples_experiment/sparse_gqa_decode_varlen_mask/sparse_gqa_decode_varlen_mask.py [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+    out_padded = output_padded.cpu()
+    out = torch.zeros((batch, heads, dim_v), dtype=out_padded.dtype, device="cpu")
+    for hid in range(heads // valid_block_H):
+        out[:, hid * valid_block_H : (hid + 1) * valid_block_H, :] = out_padded[:, hid * block_H : hid * block_H + valid_block_H, :]
+
+    return out
+
+
+# ==================== Golden (PyTorch reference) ====================
+def golden_sparse_gqa_decode_varlen_mask(query, key, value, block_mask, cache_seqlens, block_size):
```
Comment (fengz72) [inline examples_experiment/sparse_gqa_decode_varlen_mask/sparse_gqa_decode_varlen_mask.py:331]: golden相关的函数可以放到test文件里面

File: examples_experiment/sparse_gqa_decode_varlen_mask/sparse_gqa_decode_varlen_mask.py [hunk 2]
```diff
@@ -0,0 +1,425 @@
+# ruff: noqa
```
Comment (fengz72) [inline examples_experiment/sparse_gqa_decode_varlen_mask/sparse_gqa_decode_varlen_mask.py:1]: `# ruff: noqa` 全文件禁 lint, 不要使用

File: examples_experiment/sparse_gqa_decode_varlen_mask/sparse_gqa_decode_varlen_mask.py [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+import math
+
+import torch
+import torch.nn.functional as F
+
+import tilelang
+from tilelang import language as T
+
+torch.set_default_device("npu")
```
Comment (fengz72) [inline examples_experiment/sparse_gqa_decode_varlen_mask/sparse_gqa_decode_varlen_mask.py:10]: torch.set_default_device("npu") + manual_seed 在两文件模块顶层（kernel:16-17，test:460-461 且 test 里 main() 又设一遍），test 文件 import 即污染。移入入口函数。
